test/e2e/e2e_test.go

```diff
@@ -579,6 +714,17 @@ func TestE2E(t *testing.T) {
 	if err != nil {
 		t.Fatalf("error getting admin client: %v", err)
 	}
+	routeClient, err := routeclient.NewForConfig(adminConfig)
+	if err != nil {
+		t.Fatalf("error getting route client: %v", err)
+	}
+
+	// Test case to validate if metrics endpoint for the controllers are secured.
+	for _, controller := range []string{"apiservice-cabundle-injector", "configmap-cabundle-injector", "service-serving-cert-signer"} {
```

Comment:
What about the operator?

Reply:
The test cases are added in conjunction with the changes done in this PR. I can add another test case if that serves the purpose.

Reply:
A metrics endpoint is a metrics endpoint - it is exposed by the controller framework that both the operator and its controllers use. A separate test should not be required.